torch/_prims/__init__.py

```diff
@@ -1362,6 +1363,94 @@ def _broadcast_in_dim_aten(a, shape, broadcast_dimensions):
 )
 
 
+@broadcast_in_dim.py_functionalize_impl
+def _broadcast_in_dim_functionalize(ctx, a, shape, broadcast_dimensions):
+    if isinstance(ctx, CppFunctionalizeAPI):
+        return _broadcast_in_dim_functionalize_cpp(ctx, a, shape, broadcast_dimensions)
+    return _broadcast_in_dim_functionalize_python(ctx, a, shape, broadcast_dimensions)
+
+
+def _broadcast_in_dim_functionalize_python(ctx, a, shape, broadcast_dimensions):
+    (unwrapped_a,) = ctx.unwrap_tensors((a,))
+
+    def _broadcast(a_unwrapped):
+        s = list(shape)
+        dims_set = set(broadcast_dimensions)
+
+        new_strides = _compute_broadcast_in_dim_strides(a_unwrapped, s, dims_set)
+
+        # When reapply_views is disabled, we should materialize the view to avoid
+        # creating an alias. We emulate the as_strided/view behavior explicitly.
+        reapply_views = torch._C._functionalization_reapply_views_tls()
+
+        if reapply_views:
+            return torch.ops.aten.as_strided.default(
+                a_unwrapped, s, new_strides, a_unwrapped.storage_offset()
+            )
+        return torch.ops.aten.as_strided_copy.default(
+            a_unwrapped, s, new_strides, a_unwrapped.storage_offset()
+        )
+
+    with ctx.redispatch_to_next():
+        functional_broadcast = ctx.functionalize(_broadcast)
+        out = functional_broadcast(unwrapped_a)
+    return ctx.wrap_tensors(out)
+
+
+def _broadcast_in_dim_functionalize_cpp(ctx, a, shape, broadcast_dimensions):
+    (unwrapped_a,) = ctx.unwrap_tensors((a,))
+    dims_set = set(broadcast_dimensions)
+    shape_list = list(shape)
+    new_strides = _compute_broadcast_in_dim_strides(unwrapped_a, shape_list, dims_set)
+    storage_offset = unwrapped_a.storage_offset()
+
+    reapply_views = torch._C._functionalization_reapply_views_tls()
+
+    with ctx.redispatch_to_next():
+        if reapply_views:
+            out = torch.ops.aten.as_strided.default(
+                unwrapped_a, shape_list, new_strides, storage_offset
+            )
+        else:
+            out = torch.ops.aten.as_strided_copy.default(
+                unwrapped_a, shape_list, new_strides, storage_offset
+            )
+    return ctx.wrap_tensors(out)
+
+
+def _compute_broadcast_in_dim_strides(tensor, shape, broadcast_dims_set):
+    result = []
+    tensor_sizes = tensor.shape
+    tensor_strides = tensor.stride()
+
+    original_idx = 0
+    for idx in range(len(shape)):
+        if idx in broadcast_dims_set:
+            size = tensor_sizes[original_idx]
```

Comment:
can we write this to be unbacked friendly 
the unbacked semantics for broadcasting is that if we cant tell if its a broadcast case or not
we would assume no broadcasting and do a torch_check.
example this is the meta version

https://www.internalfb.com/code/fbsource/[e18e1578407a804d7877bf6be709197b739f6eae]/fbcode/caffe2/torch/_prims/__init__.py?lines=1296